src/main/java/org/checkerframework/checker/androidresource/AndroidResourceAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,402 @@
+    /**
+     * The @RawContainer annotation.
+     */
+    protected final AnnotationMirror RAW_CONTAINER =
+            AnnotationBuilder.fromClass(elements, RawContainer.class);
+    /**
+     * The @StringContainer annotation.
+     */
+    protected final AnnotationMirror STRING_CONTAINER =
+            AnnotationBuilder.fromClass(elements, StringContainer.class);
+    /**
+     * The @StyleableContainer annotation.
+     */
+    protected final AnnotationMirror STYLEABLE_CONTAINER =
+            AnnotationBuilder.fromClass(elements, StyleableContainer.class);
+    /**
+     * The @XmlContainer annotation.
+     */
+    protected final AnnotationMirror XML_CONTAINER =
+            AnnotationBuilder.fromClass(elements, XmlContainer.class);
+
+    public AndroidResourceAnnotatedTypeFactory(BaseTypeChecker checker) {
+        super(checker);
+        postInit();
+    }
+
+    @Override
+    protected TreeAnnotator createTreeAnnotator() {
+        return new ListTreeAnnotator(
+                new AndroidResourceTreeAnnotator(this), super.createTreeAnnotator());
+    }
+
+    private class AndroidResourceTreeAnnotator extends TreeAnnotator {
+
+        public AndroidResourceTreeAnnotator(AnnotatedTypeFactory annotatedTypeFactory) {
+            super(annotatedTypeFactory);
+        }
+
+        /*
+         * Method for a tree node for a member access expression. For example:
+         *       expression . identifier
+         */
+        @Override
+        public Void visitMemberSelect(MemberSelectTree node, AnnotatedTypeMirror annotatedTypeMirror) {
+
+            String expression = node.getExpression().toString();
+            String identifier = node.getIdentifier().toString();
+
+            assignContainerAnnotations(annotatedTypeMirror, expression, identifier);
+
+            assignResAnnotations(annotatedTypeMirror, node.getExpression());
+
+//            System.out.println("Expression: " + node.getExpression() + ", Identifier: " + node.getIdentifier()
+//                    + ", ATM: " + annotatedTypeMirror.toString() + "\n");
+
+            return null;
+        }
+
+        /**
+         * Assign @XXXRes annotations to member select.
+         * First it checks whether the expression has the @XXXContainer annotation
+         * then allocates the corresponding @XXXRes annotation.
+         */
+        private void assignResAnnotations(AnnotatedTypeMirror annotatedTypeMirror, ExpressionTree expressionTree) {
+
+            AnnotatedTypeMirror expressionAnnotatedTypeMirror = getAnnotatedType(expressionTree);
+
+            // Get the annotation from the {@ResourceTop} sub-hierarchy,
+            // if such an annotation targets this type; otherwise returns null.
+            AnnotationMirror expressionContainerAnnotation = expressionAnnotatedTypeMirror.getAnnotationInHierarchy(RESOURCE_TOP);
+
+            // checks if {expressionContainerAnnotation} is not null.
+            if (expressionContainerAnnotation != null) {

```

Comment:
You can reduce nesting of your code if you change this to:

```
if (xxx == null) {
  return;
}
```
That makes the code a bit easier to read.

Reply:
Fixed.